lib/matrix.rb

```diff
@@ -1750,11 +1834,39 @@ def [](i)
   alias component []
 
   def []=(i, v)
-    @elements[i]= v
+    if i.is_a?(Range)
+      raise IndexError 'expected range is outside of vector' unless in_vector_range?(i)
```

Comment:
Missing comma between IndexError and message.

Reply:
Totally missed that. Thanks a lot!